src/system/SystemPacketBuffer.cpp

```diff
@@ -530,12 +530,19 @@ PacketBufferHandle PacketBufferHandle::New(size_t aAvailableSize, uint16_t aRese
 
     CHIP_SYSTEM_FAULT_INJECT(FaultInjection::kFault_PacketBufferNew, return PacketBufferHandle());
 
-    // TODO: Change the max to a lower value
-    if (aAvailableSize > UINT32_MAX || lAllocSize > PacketBuffer::kMaxSizeWithoutReserve || lBlockSize > UINT32_MAX)
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
```

Comment:
Also, its seems weird to condition this on INET_CONFIG_ENABLE_TCP_ENDPOINT.  Seems to me like we should have CHIP_CONFIG_MAX_LARGE_PAYLOAD_SIZE_BYTES as a value.  That value might be 0 (or might be something very large?) if large payloads are not supported, and then we can have non-ifdefed code. I guess this goes to @andy31415 's comment about the logic duplication here too.  What's here right now seems really fragile and is very hard to read, which is a huge red flag for security-critical size checks.

Reply:
Defined constants in SystemPacketBuffer.h for large payload values that have been used in some of the static_asserts here.